File: Arrays & Hashing/Python/TwoSum_Leetcode-1.py
# given = [
#     nums = [2,7,11,15],
#     target = 9,
#     output = should be in index [0,1]
# ]

# Conditions = [
#     the 9 should be derived by the sum of two numbers only
#     if list is empty
# ]

nums = [2, 7, 11, 15]
target = 9


def twosum():
    length = len(nums)
    if length <= 0:
        print("List is Empty")
        return
    else:
        for num in range(length):
            for i in range(num + 1, length):
                if nums[num] + nums[i] == target:
                    print([num, i])
                    return


twosum()

File: Arrays & Hashing/Javascript/TwoSum_Leetcode-1.js
const nums = [2, 7, 11, 15];
const target = 9;

// bruteforce
function twosum(nums) {
  for (let num = 0; num < nums.length; num++) {
    for (let i = num + 1; i < nums.length; i++) {
      if (nums[num] + nums[i] === target) console.log([num, i]);
      return;
    }
  }
}

// twosum(nums);

// Hashmap

function twosumhash(nums) {
  const seen = new Map();
  for (let i = 0; i < nums.length; i++) {
    let complement = target - nums[i];
    if (seen.has(complement)) {
      console.log([seen.get(complement), i]);
      return;
    }
    seen.set(nums[i], i);
  }
  return [];
}
twosumhash(nums);
